src/Compilers/CSharp/Portable/Symbols/AssemblySymbol.cs

```diff
@@ -802,6 +811,20 @@ private NamedTypeSymbol ApplyGenericArguments(NamedTypeSymbol symbol, Type[] typ
                 return result;
             }
 
+            // Then try corlib, when finding a result there means we've found the final result
+            bool isWellKnownTypeBeforeCSharp7 = isWellKnownType && warnings is not null;
+            if (!isWellKnownTypeBeforeCSharp7 && !ignoreCorLibraryDuplicatedTypes)
```

Comment:
>!isWellKnownTypeBeforeCSharp7 && !ignoreCorLibraryDuplicatedTypes [](start = 16, length = 65)

Consider adding ```!CorLibrary.IsMissing``` check as was suggested earlier.

Reply:
Also it would be good to check that ```this``` is not a ```Corlibrary```, otherwise we already looked up in it above.

---
In reply to: [585998233](https://github.com/dotnet/roslyn/pull/51571#discussion_r585998233) [](ancestors = 585998233)